bsdbpf/bsd_bpf_sniffer.go

```diff
@@ -0,0 +1,153 @@
+// +build darwin dragonfly freebsd netbsd openbsd
+
+/*
+ * This rough sketch of a golang API for a Berkeley packet filter sniffer for BSD variants.
+ * It provides only the capability to sniff ethernet frames... but could easily be extended for
+ * sending frames as well.
+ *
+ * This file was inspired by BSD licensed code from https://github.com/songgao/ether
+ *
+ * ( http://opensource.org/licenses/BSD-3-Clause )
+ *
+ * The bpf_wordalign function was borrowed... and readFrames was very much inspired by
+ * design I saw in songgao's ether git repository; the rest of the code I wrote myself ;-)
+ *
+ * Author: David Anthony Stainton
```

Comment:
Instead of your name here, please add yourself to the AUTHORS file in the top-level directory.